File: 2962.cpp
#include <algorithm>

struct Solution {
    long long countSubarrays(vector<int>& nums, int k) {
        int n = nums.size(), maxele = *max_element(nums.begin(), nums.end()), e = 0, subCount = 0;
        long long res = 0;
        for (int s = 0; s < n; ++s){
            while (e < n && subCount < k){
                subCount += nums[e++] == maxele;
            }
            if (subCount < k){
                break;
            }
            res += n - e + 1;
            subCount -= nums[s] == maxele;
        }
        return res;
    }
};

File: sol/2962/2962.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        n = len(nums)
        maxele = max(nums)
        indices = (-1,) + tuple(i for i, x in enumerate(nums) if x == maxele)
        if len(indices) <= k:
            return 0
        res = 0
        for i in range(1, len(indices) - k + 1):
            res += (indices[i] - indices[i - 1]) * (n - indices[i + k - 1])
        return res
